### PR TITLE
cmd/dex: add dex init command

### PR DESCRIPTION
**Pull request by mhrabovcin**
JIRA: https://jira.mesosphere.com/browse/DCOS-56390

### CODE REVIEW

**Comment by mhrabovcin** [comment]
Enables: https://github.com/mesosphere/kubeaddons/pull/310

File: pkg/dex/dex.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+// UpdateConfigSecret changes config.yaml value of dex secret that is mounted
+// into dex container as a config.yaml file.
+func UpdateConfigSecret() error {
+	// load config.yaml from secret
+	c, err := utils.DefaultKubernetesClient()
+	if err != nil {
+		return err
+	}
+
+	configSecret, err := c.CoreV1().
```
Comment (shaneutt) [inline pkg/dex/dex.go]: Looks like we need to add error handling for the error produced here

File: pkg/dex/dex.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+		clientMap := staticClient.(map[string]interface{})
+		redirectURIs, found, err := unstructured.NestedStringSlice(clientMap, "redirectURIs")
+		if found && err == nil {
+			// ignore badly formatted YAML - fail somewhere else
+			for j, redirectURI := range redirectURIs {
+				parsedURI, err := url.Parse(redirectURI)
+				if err != nil {
+					return err
+				}
+				if parsedURI.Host == "PUBLIC.URI" {
```
Comment (shaneutt) [inline pkg/dex/dex.go]: Should we place this value in a constant?

File: pkg/dex/dex.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+	if err != nil {
+		return "", err
+	}
+
+	return string(secret.Data["client_secret"]), nil
+}
+
+// getOpsPortalSecretCredentials reads the credentials from the secret
+// and returns the username and password
+// FIXME: this is a temporary hack in between now and when extrasteps will be fully deprecated
```
Comment (shaneutt) [inline pkg/dex/dex.go]:
```suggestion
```

File: pkg/utils/loadbalancer.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GetServiceLoadBalancer will try to get the public LoadBalancer address, waiting for timeout
+// NOTE: This is 1:1 copy of function that already exists in `konvoy` repository
```
Comment (shaneutt) [inline pkg/utils/loadbalancer.go]: We should remove this note, it doesn't make much sense in the context of this repo.

File: pkg/utils/loadbalancer.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+// in `pkg/kubernetes`. This code should be deduped and probably removed from
+// `konvoy` repository.
+// See: https://jira.mesosphere.com/browse/DCOS-56029 for more details about
```
Comment (shaneutt) [inline pkg/utils/loadbalancer.go]: Same here, no need

**Pull request merged**